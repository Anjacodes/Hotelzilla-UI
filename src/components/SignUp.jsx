import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { registerThunk, resetCreation } from '../redux/register/registerSlice';

const SignUp = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const [form, setForm] = useState({
    name: '',
    username: '',
    email: '',
    password: '',
  });
  const [showPassword, setShowPassword] = useState(false);

  const handleChange = (e) => {
    setForm({ ...form, [e.target.name]: e.target.value });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(registerThunk(form));
  };

  const registerStatus = useSelector((store) => store.register.userCreation);
  console.log('registerStatus', registerStatus);

  useEffect(() => {
    if (registerStatus === 'fulfilled') {
      setForm({
        name: '',
        username: '',
        email: '',
        password: '',
      });

      setTimeout(() => {
        dispatch(resetCreation());
        navigate('/login', { replace: true });
      }, 3000);
    }
  }, [registerStatus]);

  if (registerStatus === 'fulfilled') {
    return (
      <div className="flex flex-col h-screen w-screen justify-center items-center">
        <section className="border rounded-md py-2 px-4">
          <h4 className="font-Taxicab text-3xl">Registration successfull!</h4>
          <p className="mt-4">You are going to be redirected to Login Screen</p>
        </section>
      </div>
    );
  }

  return (
    <div className="flex flex-col h-screen w-screen justify-center items-center">
      <section className="-mt-10">
        <header>
          <img src="Hotelzilla-logo.png" alt="" width="200" />
          <h3 className="text-center font-semibold text-2xl font-Taxicab uppercase">
            Register
          </h3>
        </header>
        <form className="flex flex-col gap-2 mt-2" onSubmit={handleSubmit}>
          <div className="flex flex-col">
            <label className="text-sm mt-1" htmlFor="name">
              Name:
            </label>
            <input
              className="border rounded px-2 py-1 hover:ring-1"
              name="name"
              id="name"
              type="text"
              value={form.name}
              onChange={handleChange}
              required
            />
          </div>

          <div className="flex flex-col">
            <label className="text-sm mt-1" htmlFor="username">
              Username:
            </label>
            <input
              className="border rounded px-2 py-1 hover:ring-1"
              name="username"
              id="username"
              type="text"
              value={form.username}
              onChange={handleChange}
              required
            />
          </div>

          <div className="flex flex-col">
            <label className="text-sm mt-1" htmlFor="email">
              Email:
            </label>
            <input
              className="border rounded px-2 py-1 hover:ring-1"
              name="email"
              id="email"
              type="email"
              value={form.email}
              onChange={handleChange}
              required
            />
          </div>

          <div className="flex flex-col">
            <label className="text-sm mt-1" htmlFor="password">
              Password:
            </label>
            <input
              className="border rounded px-2 py-1 hover:ring-1"
              name="password"
              id="password"
              type={showPassword ? 'text' : 'password'}
              value={form.password}
              onChange={handleChange}
              required
            />
          </div>

          <div className="flex justify-between items-center">
            <label className="text-sm" htmlFor="show-password">
              Show password?
            </label>
            <input
              className="hover:ring-1"
              type="checkbox"
              name="show-password"
              id="show-password"
              checked={showPassword}
              onChange={() => setShowPassword(!showPassword)}
            />
          </div>
          <button
            className="bg-green-500 w-full rounded-md py-1 font-bold text-white hover:bg-green-300 hover:text-green-800 flex px-2 items-center"
            type="submit">
            <p className="grow ml-5">Sign Up</p>
            <svg
              class={`${
                registerStatus === 'pending' ? 'text-white' : 'text-transparent'
              } animate-spin h-5 w-5`}
              xmlns="http://www.w3.org/2000/svg"
              fill="none"
              viewBox="0 0 24 24">
              <circle
                class="opacity-25"
                cx="12"
                cy="12"
                r="10"
                stroke="currentColor"
                stroke-width="4"></circle>
              <path
                class="opacity-75"
                fill="currentColor"
                d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"></path>
            </svg>
          </button>

          {registerStatus === 'rejected' && (
            <div className="bg-red-300 text-red-600 italic text-sm rounded-md px-2 py-1">
              Username not available
              <br />
              Please try a different username
            </div>
          )}
        </form>
      </section>
    </div>
  );
};

export default SignUp;
