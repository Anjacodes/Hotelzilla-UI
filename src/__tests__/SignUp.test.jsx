/**
 * @vitest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { it, describe, expect } from 'vitest';
import store from '../redux/configStore';
import SignUp from '../components/SignUp';

describe('Sign Up', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <SignUp />
        </BrowserRouter>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
