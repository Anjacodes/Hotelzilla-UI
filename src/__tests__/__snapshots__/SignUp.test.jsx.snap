// Vitest Snapshot v1

exports[`Sign Up > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-screen w-screen items-center justify-center bg-slate-100"
      >
        <div
          class="absolute top-6 left-6"
        >
          <button
            class="rounded-full border-4 p-2 text-gray-400 md:p-4 "
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowBackIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </button>
        </div>
        <form
          class="w-sm-3/4 flex flex-col gap-2 p-9"
        >
          <header
            class="flex w-full justify-center"
          >
            <div
              class="w-7/12 self-center px-5 sm:w-6/12 md:w-5/12"
            >
              <img
                alt=""
                class="self-center"
                src="Hotelzilla-logo.png"
              />
            </div>
          </header>
          <h2
            class="w-full text-center font-Taxicab text-3xl capitalize text-gray-800"
          >
            Sign up
          </h2>
          <div
            class="w-full"
          >
            <label
              for="name"
            >
              Full Name
            </label>
            <input
              class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
              id="name"
              name="name"
              placeholder="Enter your full name"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="w-full"
          >
            <label
              for="username"
            >
              Username
            </label>
            <input
              class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
              id="username"
              name="username"
              placeholder="Enter a username"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="w-full"
          >
            <label
              for="email"
            >
              Email
            </label>
            <input
              class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
              id="email"
              name="email"
              placeholder="Enter your email"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="w-full"
          >
            <label
              for="name"
            >
              Password
            </label>
            <input
              class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
              id="password"
              name="password"
              placeholder="Enter your password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="flex items-center justify-between"
          >
            <label
              class="text-sm"
              for="show-password"
            >
              Show password?
            </label>
            <input
              class="hover:ring-1"
              id="show-password"
              name="show-password"
              type="checkbox"
            />
          </div>
          <button
            class="flex items-center self-end rounded-md bg-lime-400 py-1 px-2 font-bold text-slate-50 hover:bg-lime-500 "
            type="submit"
          >
            <div
              class="ml-5 grow"
            >
              Sign Up
            </div>
            <svg
              class="text-transparent h-5 w-5 animate-spin"
              fill="none"
              viewBox="0 0 24 24"
              xmlns="http:www.w3.org/2000/svg"
            >
              <circle
                class="opacity-25"
                cx="12"
                cy="12"
                r="10"
                stroke="currentColor"
                stroke-width="4"
              />
              <path
                class="opacity-75"
                d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
                fill="currentColor"
              />
            </svg>
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-screen w-screen items-center justify-center bg-slate-100"
    >
      <div
        class="absolute top-6 left-6"
      >
        <button
          class="rounded-full border-4 p-2 text-gray-400 md:p-4 "
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </button>
      </div>
      <form
        class="w-sm-3/4 flex flex-col gap-2 p-9"
      >
        <header
          class="flex w-full justify-center"
        >
          <div
            class="w-7/12 self-center px-5 sm:w-6/12 md:w-5/12"
          >
            <img
              alt=""
              class="self-center"
              src="Hotelzilla-logo.png"
            />
          </div>
        </header>
        <h2
          class="w-full text-center font-Taxicab text-3xl capitalize text-gray-800"
        >
          Sign up
        </h2>
        <div
          class="w-full"
        >
          <label
            for="name"
          >
            Full Name
          </label>
          <input
            class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
            id="name"
            name="name"
            placeholder="Enter your full name"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="w-full"
        >
          <label
            for="username"
          >
            Username
          </label>
          <input
            class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
            id="username"
            name="username"
            placeholder="Enter a username"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="w-full"
        >
          <label
            for="email"
          >
            Email
          </label>
          <input
            class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
            id="email"
            name="email"
            placeholder="Enter your email"
            required=""
            type="email"
            value=""
          />
        </div>
        <div
          class="w-full"
        >
          <label
            for="name"
          >
            Password
          </label>
          <input
            class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
            id="password"
            name="password"
            placeholder="Enter your password"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="flex items-center justify-between"
        >
          <label
            class="text-sm"
            for="show-password"
          >
            Show password?
          </label>
          <input
            class="hover:ring-1"
            id="show-password"
            name="show-password"
            type="checkbox"
          />
        </div>
        <button
          class="flex items-center self-end rounded-md bg-lime-400 py-1 px-2 font-bold text-slate-50 hover:bg-lime-500 "
          type="submit"
        >
          <div
            class="ml-5 grow"
          >
            Sign Up
          </div>
          <svg
            class="text-transparent h-5 w-5 animate-spin"
            fill="none"
            viewBox="0 0 24 24"
            xmlns="http:www.w3.org/2000/svg"
          >
            <circle
              class="opacity-25"
              cx="12"
              cy="12"
              r="10"
              stroke="currentColor"
              stroke-width="4"
            />
            <path
              class="opacity-75"
              d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
              fill="currentColor"
            />
          </svg>
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SignUp > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-screen w-screen items-center justify-center bg-slate-100"
      >
        <div
          class="absolute top-6 left-6"
        >
          <button
            class="rounded-full border-4 p-2 text-gray-400 md:p-4 "
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowBackIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </button>
        </div>
        <form
          class="w-sm-3/4 flex flex-col gap-2 p-9"
        >
          <header
            class="flex w-full justify-center"
          >
            <div
              class="w-7/12 self-center px-5 sm:w-6/12 md:w-5/12"
            >
              <img
                alt=""
                class="self-center"
                src="Hotelzilla-logo.png"
              />
            </div>
          </header>
          <h2
            class="w-full text-center font-Taxicab text-3xl capitalize text-gray-800"
          >
            Sign up
          </h2>
          <div
            class="w-full"
          >
            <label
              for="name"
            >
              Full Name
            </label>
            <input
              class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
              id="name"
              name="name"
              placeholder="Enter your full name"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="w-full"
          >
            <label
              for="username"
            >
              Username
            </label>
            <input
              class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
              id="username"
              name="username"
              placeholder="Enter a username"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="w-full"
          >
            <label
              for="email"
            >
              Email
            </label>
            <input
              class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
              id="email"
              name="email"
              placeholder="Enter your email"
              required=""
              type="email"
              value=""
            />
          </div>
          <div
            class="w-full"
          >
            <label
              for="name"
            >
              Password
            </label>
            <input
              class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
              id="password"
              name="password"
              placeholder="Enter your password"
              required=""
              type="password"
              value=""
            />
          </div>
          <div
            class="flex items-center justify-between"
          >
            <label
              class="text-sm"
              for="show-password"
            >
              Show password?
            </label>
            <input
              class="hover:ring-1"
              id="show-password"
              name="show-password"
              type="checkbox"
            />
          </div>
          <button
            class="flex items-center self-end rounded-md bg-lime-400 py-1 px-2 font-bold text-slate-50 hover:bg-lime-500 "
            type="submit"
          >
            <div
              class="ml-5 grow"
            >
              Sign Up
            </div>
            <svg
              class="text-transparent h-5 w-5 animate-spin"
              fill="none"
              viewBox="0 0 24 24"
              xmlns="http:www.w3.org/2000/svg"
            >
              <circle
                class="opacity-25"
                cx="12"
                cy="12"
                r="10"
                stroke="currentColor"
                stroke-width="4"
              />
              <path
                class="opacity-75"
                d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
                fill="currentColor"
              />
            </svg>
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-screen w-screen items-center justify-center bg-slate-100"
    >
      <div
        class="absolute top-6 left-6"
      >
        <button
          class="rounded-full border-4 p-2 text-gray-400 md:p-4 "
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
        </button>
      </div>
      <form
        class="w-sm-3/4 flex flex-col gap-2 p-9"
      >
        <header
          class="flex w-full justify-center"
        >
          <div
            class="w-7/12 self-center px-5 sm:w-6/12 md:w-5/12"
          >
            <img
              alt=""
              class="self-center"
              src="Hotelzilla-logo.png"
            />
          </div>
        </header>
        <h2
          class="w-full text-center font-Taxicab text-3xl capitalize text-gray-800"
        >
          Sign up
        </h2>
        <div
          class="w-full"
        >
          <label
            for="name"
          >
            Full Name
          </label>
          <input
            class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
            id="name"
            name="name"
            placeholder="Enter your full name"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="w-full"
        >
          <label
            for="username"
          >
            Username
          </label>
          <input
            class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
            id="username"
            name="username"
            placeholder="Enter a username"
            required=""
            type="text"
            value=""
          />
        </div>
        <div
          class="w-full"
        >
          <label
            for="email"
          >
            Email
          </label>
          <input
            class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
            id="email"
            name="email"
            placeholder="Enter your email"
            required=""
            type="email"
            value=""
          />
        </div>
        <div
          class="w-full"
        >
          <label
            for="name"
          >
            Password
          </label>
          <input
            class="focus:shadow-outline w-full appearance-none rounded border py-2 px-3 leading-tight text-gray-700 focus:outline-none"
            id="password"
            name="password"
            placeholder="Enter your password"
            required=""
            type="password"
            value=""
          />
        </div>
        <div
          class="flex items-center justify-between"
        >
          <label
            class="text-sm"
            for="show-password"
          >
            Show password?
          </label>
          <input
            class="hover:ring-1"
            id="show-password"
            name="show-password"
            type="checkbox"
          />
        </div>
        <button
          class="flex items-center self-end rounded-md bg-lime-400 py-1 px-2 font-bold text-slate-50 hover:bg-lime-500 "
          type="submit"
        >
          <div
            class="ml-5 grow"
          >
            Sign Up
          </div>
          <svg
            class="text-transparent h-5 w-5 animate-spin"
            fill="none"
            viewBox="0 0 24 24"
            xmlns="http:www.w3.org/2000/svg"
          >
            <circle
              class="opacity-25"
              cx="12"
              cy="12"
              r="10"
              stroke="currentColor"
              stroke-width="4"
            />
            <path
              class="opacity-75"
              d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
              fill="currentColor"
            />
          </svg>
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
